app/kubernetes-ingress-controller/1.0.x/concepts/ingress-versions.md

```diff
@@ -0,0 +1,163 @@
+
+#### Prefix
+
+If `pathType` is `Prefix`, the controller creates a Kong route with two path
+criteria, e.g. `/foo` will create a route with a `/foo$` regular expression and
+`/foo/` plain path.
+
+#### ImplementationSpecific
+
+The controller leaves ImplementationSpecific path rules entirely up to the Kong
+router. It creates a route with the exact same path string as the Ingress rule.
+
+## Ingress class
+
+[Ingress class][ingress-class] indicates which resources an ingress controller
+should process. It provides a means to separate out configuration intended for
+other controllers or other instances of the Kong Ingress Controller.
+
+In v1beta1, ingress class was handled informally using
+`kubernetes.io/ingress.class` [annotations][deprecated-annotation]. v1
+introduces a new [IngressClass resource][ingress-class-api] which provides
+richer information about the controller. v1 Ingresses are bound to a class via
+their `ingressClassName` field.
+
+For example, consider this v1beta1 Ingress:
+
+```
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: example-ingress
+  annotations:
+    kubernetes.io/ingress.class: "kong"
+spec:
+  rules:
+  - host: example.com
+    http:
+      paths:
+      - path: /test
+        backend:
+          serviceName: echo
+          servicePort: 80
+```
+
+Its ingress class annotation is set to `kong`, and ingress controllers set to
+process `kong` class Ingresses will process it.
+
+In v1, the equivalent configuration declares a `kong` IngressClass resource
+whose `metadata.name` field indicates the class name. The `ingressClassName`
+value of the Ingress object must match the value of the `name` field in the
+IngressClass metadata:
+
+```
+apiVersion: networking.k8s.io/v1
+kind: IngressClass
+metadata:
+  name: kong
+spec:
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: example-ingress
+spec:
+  ingressClassName: kong
+  rules:
+  - http:
+      paths:
+      - path: /testpath
+        pathType: Prefix
+        backend:
+          service:
+            name: test
+            port:
+              number: 80
+```
+
+## Hostnames
+
+Ingress v1 formally codifies support for [wildcard hostnames][wildcard-hostnames].
+v1beta1 Ingresses did not reject wildcard hostnames, however, and Kong had
+[existing support for them][kong-wildcard-hostnames].
+
+As such, while the v1beta1 specification did not officially support wildcard
+hostnames, you can use wildcard hostnames with either version. Setting a
+hostname like `*.example.com` will match requests for both `foo.example.com`
+and `bar.example.com` with either v1 or v1beta1 Ingresses.
+
+## Backend types
+
+Ingress v1 introduces support for backends other than Kubernetes Services, via
+[resource backends][resource-backends].
+
+Kong has support for routes without a service in some cases, such as routes
+that use the [AWS Lambda plugin][lambda-plugin]. However, it does not yet offer
```

Comment:
I don't know if it's a good idea to phrase something as "does not yet do this thing", since that sounds like you're promising a feature change in the future. Whether this feature is going to be available or not is not super relevant - timelines and roadmaps change, so I'd be wary of any pseudo-promises.

Reply:
Yeah--we're in a weird position where we kinda supported something akin to resource backends through our existing features, and the question of how to write K8S configuration for those even though there's no real appropriate Service definition does come up in practice (recently https://kongstrong.slack.com/archives/CA42V003B/p1606771633206900).

We need to cover the workaround that enables those use cases (dummy Services) somewhere while acknowledging their relation to the new concept introduced in Ingress v1 that we don't support. Not sure where that will go in the future (since K8S' own Service APIs roadmap means many current Ingress concepts will change significantly); have reworded to say that we don't support any service backend stuff, but that you should use the dummy Services where appropriate.